### ISSUE TITLE
doc/articles/wiki: multiple issues

### ISSUE THREAD

**Issue by adregan** [golang/go#9566]
When running both the code I've written and the final output given at the end of the tutorial, the edit pages throw the following error:

```
http: panic serving [::1]:60627: runtime error: invalid memory address or nil pointer dereference
```

Furthermore, the tutorial completely skips over the import of `"flag"` and `"log"` and if you follow along closely, you cannot build at the end as this block :

```go
    if *addr {
        l, err := net.Listen("tcp", "127.0.0.1:0")
        if err != nil {
            log.Fatal(err)
        }
        err = ioutil.WriteFile("final-port.txt", []byte(l.Addr().String()), 0644)
        if err != nil {
            log.Fatal(err)
        }
        s := &http.Server{}
        s.Serve(l)
        return
    }
```

throws several errors due to missing imports as mentioned above, but also as `addr` isn't addressed in the tutorial but does appear in the final version 

```go
var (
    addr = flag.Bool("addr", false, "find open address and print to final-port.txt")
)
```

A syntax, which if you are new to go (like me), isn't mentioned at all.

**Comment by adregan** [comment]
Good news. The panic error was in the template. However, I'd still like to know more about the log and flag imports

**Comment by zombiezen** [comment]
This is fixed as of 44618b28d496b281c3966742eacfe3618f060d21: https://tip.golang.org/doc/articles/wiki/
